### PR TITLE
Cai Nowicki Unit 2 pull

### PR DESCRIPTION
**Pull request by CaiNowicki**

### CODE REVIEW

**Comment by llpk79** [comment]
### Assignment
- [x] Look at the data. Choose a feature, and plot its relationship with the target.
    - Great work getting a baseline!
    - Try using the `opacity` argument in the plotly graph to make that graph look not quite so odd.
- [x] Use scikit-learn for linear regression with one feature. You can follow the 5-step process from Jake VanderPlas.
    - Great job comparing your results to the baseline!
    - Make sure your notebooks run without errors unless you're making a point with the error.
- [x] Define a function to make new predictions and explain the model coefficient.
    - How can you make this function more versatile? Rewrite it to take any valid column name.
- [x] Organize and comment your code.
    - Good job. Consider making code comments in `# this is what this code does blocks` and comments about code results in markdown for a clean look.
### Stretch Goals
- [x] Do linear regression with two or more features.
    - Nice job exploring engineered features!
    - Got a decent result with beds, baths and `features_total` 😄

**Comment by llpk79** [comment]
## Regression 2
- [x] Do train/test split. Use data from April & May 2016 to train. Use data from June 2016 to test.
- [x] Engineer at least two new features. (See below for explanation & ideas.)
    - `train['pets_allowed'] = train['dogs_allowed'] & train['cats_allowed']`
- [x] Fit a linear regression model with at least two features.
    - Love the exploration!
- [x] Get the model's coefficients and intercept.
    - Sure nuf.
- [x] Get regression metrics RMSE, MAE, and R^2, for both the train and test data.
    - The error you get when guessing the mean is the baseline. not the mean itself.
- [x] What's the best test MAE you can get? Share your score and features used with your cohort on Slack!
    - Not bad at all!!
- [x] As always, commit your notebook to your fork of the GitHub repo.
    - ✔️

**Comment by llpk79** [comment]
## Ridge Regression
- [x] Do train/test split. Use data from January — March 2019 to train. Use data from April 2019 to test.
    - Good call to drop the columns with high instance of nan in this case. Easement may be important if dealt with properly.
    - How do you know your data is from Jan - March?
- [x] Do one-hot encoding of categorical features.
    - Kinda weird that `LAND_SQUARE_FEET` is in the categorical columns...
- [x] Do feature selection with `SelectKBest`.
- [x] Do [feature scaling](https://scikit-learn.org/stable/modules/preprocessing.html).
- [x] Fit a ridge regression model with multiple features.
- [x] Get mean absolute error for the test set.
- [x] As always, commit your notebook to your fork of the GitHub repo.
    - ✔️ 

Nice work, Cai.

**Comment by llpk79** [comment]
## Logistic Regression
- [x] Do train/validate/test split. Train on reviews from 2016 & earlier. Validate on 2017. Test on 2018 & later.
    - Nicely done.
    - Great work cleaning up this data!
- [x] Begin with baselines for classification.
    - Nice!
- [x] Use scikit-learn for logistic regression.
    - Way to write DRY code by writing a function!
- [x] Get your model's validation accuracy. (Multiple times if you try multiple iterations.)
    - Easy to do it every time when it's built into a function.
- [x] Get your model's test accuracy. (One time, at the end.)
    - That's a great result! We can usually expect our model to do a bit worse on the test set than we saw as our best with the validation set, partly because, as we tweak things to get a better validation score, we tend to overfit to the validations set.
- [x] Commit your notebook to your fork of the GitHub repo.
    - ✔️ 
    
## Stretch Goals
- [ ] Add your own stretch goal(s) !
- [ ] Make exploratory visualizations.
- [ ] Do one-hot encoding.
- [ ] Do [feature scaling](https://scikit-learn.org/stable/modules/preprocessing.html).
- [ ] Get and plot your coefficients.
- [ ] Try [scikit-learn pipelines](https://scikit-learn.org/stable/modules/compose.html).
